csrc/root_domain_map.cpp

```diff
@@ -121,6 +121,34 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   const auto producer_root =
       TensorDomain::noReductions(producer->maybeRFactor());
   const auto& consumer_root = consumer->root();
+
+  // Map MatmulOp extents in reverse
+  if (consumer_tv_->definition()->isA<MatmulOp>()) {
+    int ritc = consumer_root.size() - 1;
+    int ritp = producer_root.size() - 1;
+
+    while (ritc >=0 && ritp >= 0) {
+      IterDomain* producer_id = producer_root.at(ritp);
+      IterDomain* consumer_id = consumer_root.at(ritc);
+      if (!producer_id->extent()->sameAs(consumer_id->extent())){
```

Comment:
This is to avoid mapping K to either M/N in the consumer. Tests pass for symbolic shapes, however, how are extents checked for symbolic shapes? Have we already concretized at this point?

Reply:
I don't think this is safe to do. We might have a different extent here when the inputs are `Symbolic`. Instead, I think we need to use the same logic as in the `matmul` op here to determine whether these axes correspond. If possible, it would be nice to factor out that logic from the op so that it can be used here.